### DISCUSSION TITLE
`The handle is invalid` when the a CLI application needs console.

### DISCUSSION THREAD

**Discussion by qui8t** [xunit/xunit#2478] [Question] [answered]
In various part of my console application, I've calls such as the following.

```csharp
Console.CursorVisible = false;
Console.SetCursorPosition(Left, Top);
```

When testing my application using Xunit, and when calls such as the above are made, it throws the following exception:

```
System.IO.IOException: 'The handle is invalid.'
```

I'm not testing whether the cursor is set to a particular position or not, merely, such console calls are made within the methods I'm testing for another functionality.

Since I cannot remove any of such calls (because they are required to "properly" display the output of my application), my current workaround is as the following; however, since this workaround convolutes the code, it is not a proper fix. 

```csharp
try { Console.CursorVisible = true; }
catch (IOException) { }
```

**Comment by bradwilson** [comment, marked as answer]
Can you provide more information, including a repro project? I tried simply doing `Console.CursorVisible = true;` in a unit test and didn't have any exceptions being thrown. I tested with .NET Framework 4.7.2 (both via `xunit.console` and `dotnet test`) as well as .NET 6.0 (via `dotnet test`).

**Reply by bradwilson** [reply]
Please open an issue with a repro project. Thanks!

**Reply by bradwilson** [reply]
I'm going to guess that the issue is that we launch your test process with redirected console input/output, which means there isn't an actual console, and that's causing the problem. If that's the case, then there's really not anything we can do to fix it (since this is required in the design of v3).

https://github.com/xunit/xunit/blob/cadd74d0cf2f67e1d73a784cd7148e7957da09f3/src/xunit.v3.runner.utility/Frameworks/v3/LocalTestProcess.cs#L119-L124

**Comment by mbarbac** [comment]
Has there been any progress here?

**Reply by bradwilson** [reply]
Nobody has provided a repro project, so no.

**Comment by mbarbac** [comment]
Almost any Console-related operation inside a test method will do. For instance, just add 'Console.CursorTop=1' to a test method and you'll get a 'System.IO.IOException' with a 'handle is invalid' message. This is the easiest example I could made. I'm using W11, latest Visual Studio, and the standard Xunit template with the latest Xunit package updates. Hope this helps.

**Comment by mbarbac** [comment]
Hummm, not really, no redirection: just a brand new test project using the VS template, updated Xunit packages, and a single test method in a single test case.

Moisés Barba Cebeira
***@***.***
________________________________

**Reply by bradwilson** [reply]
> Hummm, not really, no redirection

It's not your code, it's our code.

**Comment by mbarbac** [comment]
Ok, understood, fair enough. You’ve been very kind answering, thanks a lot. But may I then kindly suggest making this fact (ie: “cannot use Xunit with Console”) explicit, so that others can avoid this scenario?
